### ISSUE TITLE
Include index information

### ISSUE THREAD

**Issue by linawolf** [TYPO3-Documentation/t3docs-search-indexer#33]
In the larger docs we have a manually created index: 
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/genindex.html

Would it be possible to take it into account when doing searches? For example I currently get not search results for 'AdditionalConfiguration' even though it can be found in the index.

**Comment by tmotyl** [comment]
Do you want to make index page show in the search results page? I don't think it's useful, as it will probably be always very high.
However we could think about using the keywords used in index and enrich the target page with them, what do you think?

**Comment by linawolf** [comment]
Yes, the latter is what would be great: use the manual index to enrich the target page.

**Comment by tmotyl** [comment]
how are these keywords visible in html? The search indexer parses the html.

**Comment by linawolf** [comment]
They are currently not displayed. @marble would it be possible to display the indexes on the page itself?

**Comment by tmotyl** [comment]
@linawolf the AdditionalConfiguration case will be solved with https://github.com/TYPO3-Documentation/t3docs-search-indexer/pull/32

**Comment by linawolf** [comment]
Nevertheless I think it would be a great tool to include indexes. As we already put much brainpower into them and can use synonyms and such without cluttering the text

**Comment by tmotyl** [comment]
Yes I agree that including inxed would be nice.

**Comment by tmotyl** [comment]
another test example:
$GLOBALS[TCA]

![image](https://github.com/TYPO3-Documentation/t3docs-search-indexer/assets/515397/f38e9d7d-9e64-41fe-8116-c06f19838024)

**Comment by soee** [comment]
I have talked with @linawolf  on Slack about this task and what we want to achieve here.

First of all, we do not want to index the `genindex.html` file and its content. It has been excluded from indexing so far, and we will keep it this way.

We do want to have some way to enrich each section of an article with keywords, and such keywords should be indexed and then boost search results if they match words from the searched phrase.

The proposed solution, which we have agreed upon, would be implemented for the new rendering only.

- while generating documentation files, each section tag inside the article tag could have the `data-keywords` attribute set.
- this `data-keywords` attribute as a value would have a comma-separated list of some keywords.
- while indexing each section (treated as a manual snippet which we can search for), these keywords would be indexed - 
- inside the keywords field, which we introduced already with #35.
- indexed keywords would boost a snippet in search results if any of them matches a word from the search phrase.
